File: src/main/java/org/telran/lecture_03_recursion/examples/indirectRecursion.js
function functionA(number) {
    console.log("functionA: " + number);
    if (number > 0) {
        functionB(number - 1);
    }
}

function functionB(number) {
    console.log("functionB: " + number);
    if (number > 0) {
        functionA(number - 1);
    }
}

functionA(3);

File: src/main/java/org/telran/lecture_03_recursion/examples/IndirectRecursion.java
package org.telran.lecture_03_recursion.examples;

public class IndirectRecursion {
    public static void main(String[] args) {
        functionA(3);
    }


    public static void functionA(int number) {
        System.out.println("functionA: " + number);
        if (number != 5) {
            functionB(number - 1);
        }
    }

    public static void functionB(int number) {
        System.out.println("functionB: " + number);
        if (number != 5) {
            functionA(number - 1);
        }
    }


}
